packages/models-library/src/models_library/api_schemas_webserver/functions_wb_schema.py

```diff
@@ -0,0 +1,276 @@
+from collections.abc import Mapping
+from enum import Enum
+from typing import Annotated, Any, Literal, TypeAlias
+from uuid import UUID
+
+from common_library.errors_classes import OsparcErrorMixin
+from models_library import projects
+from models_library.services_types import ServiceKey, ServiceVersion
+from pydantic import BaseModel, Field
+
+from ..projects import ProjectID
+
+FunctionID: TypeAlias = UUID
+FunctionJobID: TypeAlias = UUID
+FileID: TypeAlias = UUID
+
+InputTypes: TypeAlias = FileID | float | int | bool | str | list
+
+
+class FunctionSchemaClass(str, Enum):
+    json_schema = "application/schema+json"
+
+
+class FunctionSchemaBase(BaseModel):
+    schema_content: Any = Field(default=None)
+    schema_class: FunctionSchemaClass
+
+
+class JSONFunctionSchema(FunctionSchemaBase):
+    schema_content: Mapping[str, Any] = Field(
+        default={}, description="JSON Schema", title="JSON Schema"
+    )  # json-schema library defines a schema as Mapping[str, Any]
+    schema_class: FunctionSchemaClass = FunctionSchemaClass.json_schema
+
+
+class JSONFunctionInputSchema(JSONFunctionSchema):
+    schema_class: Literal[FunctionSchemaClass.json_schema] = (
+        FunctionSchemaClass.json_schema
+    )
+
+
+class JSONFunctionOutputSchema(JSONFunctionSchema):
+    schema_class: Literal[FunctionSchemaClass.json_schema] = (
+        FunctionSchemaClass.json_schema
+    )
+
+
+FunctionInputSchema: TypeAlias = Annotated[
+    JSONFunctionInputSchema,
+    Field(discriminator="schema_class"),
+]
+
+FunctionOutputSchema: TypeAlias = Annotated[
```

Comment:
minor: I think in osparc we usually have capital letters enums.

Reply:
done.